src/Analyzers/CSharp/Tests/RemoveUnnecessaryDiscardDesignation/RemoveUnnecessaryDiscardDesignationTests.cs

```diff
@@ -0,0 +1,276 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.RemoveUnnecessaryDiscardDesignation;
+using Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.RemoveUnnecessaryDiscardDesignation
+{
+    using VerifyCS = CSharpCodeFixVerifier<
+        CSharpRemoveUnnecessaryDiscardDesignationDiagnosticAnalyzer,
+        RemoveUnnecessaryDiscardDesignationCodeFixProvider>;
+
+    public class RemoveUnnecessaryDiscardDesignationTests
+    {
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryDiscardDesignation)]
+        public async Task TestDeclarationPatternInSwitchStatement()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+class C
+{
+    void M(object o)
+    {
+        switch (o)
+        {
+            case int [|_|]:
+                break;
+        }
+    }
+}",
+                FixedCode = @"
+class C
+{
+    void M(object o)
+    {
+        switch (o)
+        {
+            case int:
```

Comment:
I think there needs to be a follow-up issue/PR to fix another existing IDE code fix (removing unnecessary locals/parameters). For example, https://github.com/dotnet/roslyn/blob/dfb8f6351ac881f133e54b3e3a3f3801e859c01d/src/Analyzers/CSharp/Tests/RemoveUnusedParametersAndValues/RemoveUnusedValueAssignmentTests.cs#L2404-L2432

This code fix should be modified to generate `case int:` instead of `case int _:`. @CyrusNajmabadi can you please open a new tracking issue for it?

Reply:
Done: https://github.com/dotnet/roslyn/issues/49155